File: javascript/easy/group-anagrams.js
var groupAnagrams = function(strs) {
    let anagram = {};

        for(let str of strs){
            let count = new Array(26).fill(0);
            for(let char of str){
                count[char.charCodeAt(0) - "a".charCodeAt(0)] += 1;
            }
            let key = count.join(",");
            if(!anagram[key]){
                anagram[key] = [];
            }
            anagram[key].push(str);
        }

        return Object.values(anagram)
};

File: python/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramMap = defaultdict(list)

        for string in strs:
            count = [0] * 25
            for letter in string:
                count[ord(letter) - ord("a")] += 1
            anagramMap[tuple(count)].append(string)
        return list(anagramMap.values())
